### ISSUE TITLE
loss 计算部分的优化疑问

### ISSUE THREAD

**Issue by mx8435** [Oneflow-Inc/one-glm#23]
我看了下`mpu.vocab_parallel_cross_entropy`实现上有涉及到不同workers之间通信，而改为`losses = flow._C.sparse_softmax_cross_entropy(logits, labels)`后 ，这个api内部有不同模型并行部分的worker进行logits all_reduce吗？还是说这个只能用于非模型并行的情况？
@BBuf

**Comment by BBuf** [comment]
> 我看了下`mpu.vocab_parallel_cross_entropy`实现上有涉及到不同workers之间通信，而改为`losses = flow._C.sparse_softmax_cross_entropy(logits, labels)`后 ，这个api内部有不同模型并行部分的worker进行logits all_reduce吗？还是说这个只能用于非模型并行的情况？ @BBuf

这个api内部会自动管理模型并行的通信，不过目前我们没有为这个仓库适配模型并行，暂时只支持数据并行。我们会在libai仓库里支持glm的模型并行以及流水并行
